bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/BufferedReadChannel.java

```diff
@@ -95,9 +98,10 @@ public synchronized int read(ByteBuf dest, long pos, int length) throws IOExcept
                     throw new IOException("Reading from filechannel returned a non-positive value. Short read.");
                 }
                 readBuffer.writerIndex(readBytes);
+                bytesFetchedFromFile += readBytes;
             }
         }
-        return (int) (currentPosition - pos);
+        return new ImmutablePair<> ((int) (currentPosition - pos), bytesFetchedFromFile);
```

Comment:
This is a very frequently called code, can we avoid creating a new object?
I think the throttler can be safely passed down the call chain into this method.

Reply:
Good catch. I'm considering how to avoiding creating new object. 
The `BufferedReadChannel.read` method is synchronized, we should use throttle in the read method to block it when throttled, otherwise, it will block other threading calling.  I still have no good idea about it.

Reply:
@hangc0276 @dlg99 
I think you can write like this simple code :
`return Pair.of((int) (currentPosition - pos), bytesFetchedFromFile); `
instead of 
`return new ImmutablePair<> ((int) (currentPosition - pos), bytesFetchedFromFile); `

but I think it's ok, frequent creation of ImmutablePair has no affect 
<img width="964" alt="image" src="https://user-images.githubusercontent.com/42990025/181050024-667fd915-ba55-4de9-9bb1-da395b60391e.png">